src/lang/modifyAst/addFillet.ts

```diff
@@ -28,26 +29,151 @@ import {
+
+  // 2. get path
+  const programMemory = kclManager.programMemory
+  const artifactGraph = engineCommandManager.artifactGraph
+  const getPathToExtrudeForSegmentSelectionResult =
+    getPathToExtrudeForSegmentSelection(
+      ast,
+      selection,
+      programMemory,
+      artifactGraph
+    )
+  if (err(getPathToExtrudeForSegmentSelectionResult))
+    return getPathToExtrudeForSegmentSelectionResult
+  const { pathToSegmentNode, pathToExtrudeNode } =
+    getPathToExtrudeForSegmentSelectionResult
+
+  // 3. add fillet
+  const addFilletResult = addFillet(
+    ast,
+    pathToSegmentNode,
+    pathToExtrudeNode,
+    'variableName' in radius ? radius.variableIdentifierAst : radius.valueAst
+  )
+  if (trap(addFilletResult)) return addFilletResult
+  const { modifiedAst, pathToFilletNode } = addFilletResult
+
+  // 4. update ast
+  updateAstAndFocus(modifiedAst, pathToFilletNode)
+}
+
+function getAst(
+  ast: Program,
+  radius: KclCommandValue
+): { ast: Program } | Error {
+  try {
+    // Validate and update AST
+    if (
+      'variableName' in radius &&
+      radius.variableName &&
+      radius.insertIndex !== undefined
+    ) {
+      const newBody = [...ast.body]
+      newBody.splice(radius.insertIndex, 0, radius.variableDeclarationAst)
+      ast.body = newBody
+    }
+    return { ast }
+  } catch (error) {
+    return new Error(`Failed to handle AST: ${(error as Error).message}`)
+  }
+}
+
+export function getPathToExtrudeForSegmentSelection(
+  ast: Program,
+  selection: Selections,
+  programMemory: ProgramMemory,
+  artifactGraph: ArtifactGraph
+): { pathToSegmentNode: PathToNode; pathToExtrudeNode: PathToNode } | Error {
+  const pathToSegmentNode = getNodePathFromSourceRange(
+    ast,
+    selection.codeBasedSelections[0].range
+  )
+
+  const varDecNode = getNodeFromPath<VariableDeclaration>(
+    ast,
+    pathToSegmentNode,
+    'VariableDeclaration'
+  )
+  if (err(varDecNode)) return varDecNode
+  const sketchVar = varDecNode.node.declarations[0].id.name
+
+  const sketchGroup = programMemory.get(sketchVar)
+  if (sketchGroup?.type !== 'SketchGroup')
+    return new Error('Invalid sketch group type')
+
+  const extrusion = getExtrusionFromSuspectedPath(sketchGroup.id, artifactGraph)
+
+  const pathToExtrudeNode = err(extrusion)
+    ? []
+    : getNodePathFromSourceRange(ast, extrusion.codeRef.range)
+  return { pathToSegmentNode, pathToExtrudeNode }
+}
+
+async function updateAstAndFocus(
+  modifiedAst: Program,
+  pathToFilletNode: PathToNode
+) {
+  const updatedAst = await kclManager.updateAst(modifiedAst, true, {
+    focusPath: pathToFilletNode,
+  })
+  if (updatedAst?.selections) {
+    editorManager.selectRange(updatedAst?.selections)
+  }
+}
+
+/**
+ * Add Fillet
+ */
 
 export function addFillet(
   node: Program,
   pathToSegmentNode: PathToNode,
   pathToExtrudeNode: PathToNode,
+  // radius = createLiteral(5) as Expr // TODO what is EXP?
   radius = createLiteral(5) as Expr
```

Comment:
thanks, good point.. I've used `radius: Expr = createLiteral(5)` . Just deleting `as Expr` did not work because ts is not sure whether radius is a `Literal` in this line: `'variableName' in radius ? radius.variableIdentifierAst : radius.valueAst` . efbc67878bc4ca485d087aa9f6e2cf4f803cdc01